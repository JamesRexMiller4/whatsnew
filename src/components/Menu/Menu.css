nav {
    display: flex;
    justify-content: space-around
}

.btn {
    width: 20%;
    height: 40px;
    background-color: rgb(0, 115, 255);
    border: 5px solid;
    font-size: 15px;
}

.btn:hover {
    background-color: aquamarine;
    cursor: pointer;
    color: black;
    font-size: 20px;
    font-weight: bold;
}