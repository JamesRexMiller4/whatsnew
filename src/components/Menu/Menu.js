import React from 'react';
import './Menu.css'

const Menu = () => {
    return(
        <aside>
            <
        </aside>
    )
}

export default Menu;