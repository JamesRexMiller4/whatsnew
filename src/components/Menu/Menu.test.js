import React from 'react';
import Menu from './Menu';
import {shallow} from 'enzyme';
import local from '../../data/local';

describe('Menu', () => {
    let wrapper, chooseNewsMock;

    beforeEach(() => {
        chooseNewsMock = jest.fn()
        wrapper = shallow(<Menu 
        chooseNews={chooseNewsMock}
        />)
    })
    it('should match the snapshot with all the data passed correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    it('should trigger the chooseNews method when a button is clicked', () => {
        wrapper.find('#local').simulate('click');
        expect(chooseNewsMock).toHaveBeenCalledWith('local');
    })
})