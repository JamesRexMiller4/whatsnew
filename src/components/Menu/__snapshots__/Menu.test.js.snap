// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot with all the data passed correctly 1`] = `
<nav>
  <button
    className="btn"
    id="local"
    onClick={[Function]}
    type="button"
  >
    Local News
  </button>
  <button
    className="btn"
    id="technology"
    onClick={[Function]}
    type="button"
  >
    Technology
  </button>
  <button
    className="btn"
    id="entertainment"
    onClick={[Function]}
    type="button"
  >
    Entertainment
  </button>
  <button
    className="btn"
    id="science"
    onClick={[Function]}
    type="button"
  >
    Science
  </button>
  <button
    className="btn"
    id="health"
    onClick={[Function]}
    type="button"
  >
    Health
  </button>
</nav>
`;
